invoice/src/main/java/org/killbill/billing/invoice/api/InvoiceApiHelper.java

```diff
@@ -103,7 +109,9 @@ public List<InvoiceItem> dispatchToInvoicePluginsAndInsertItems(final UUID accou
             final List<InvoiceModelDao> invoiceModelDaos = new LinkedList<InvoiceModelDao>();
             for (final DefaultInvoice invoiceForPlugin : invoicesForPlugins) {
                 // Call plugin(s)
-                invoicePluginDispatcher.updateOriginalInvoiceWithPluginInvoiceItems(invoiceForPlugin, isDryRun, context, priorCallResult.getPluginProperties(), internalCallContext);
+                final AdditionalInvoiceItemsResult itemsResult = invoicePluginDispatcher.updateOriginalInvoiceWithPluginInvoiceItems(invoiceForPlugin, isDryRun, context, pluginProperties, internalCallContext);
+                // Could be a bit weird for a plugin to keep updating properties for each invoice
```

Comment:
FWIW that only works because the plugins are called in sequence for each invoice. In other words, for each invoice iteration, the first plugin would add the right plugin properties for taxation (based on each invoice item id) for the next invoice plugin, before moving on to the next invoice.

With this in mind, wondering then if we need to _clear_ the adjusted properties between each invoice?

Reply:
The model is to rely the union of all the properties (input + adjusted). I don't think the Kill Bill core should clear anything, but the plugin could of course decide to clear things (if it knows the consequences)